docs/content.zh/release-notes/flink-1.14.md

```diff
@@ -0,0 +1,424 @@
+
+#### Deprecate toAppendStream and toRetractStream
+
+##### [FLINK-23330](https://issues.apache.org/jira/browse/FLINK-23330)
+
+The outdated variants of `StreamTableEnvironment.{fromDataStream|toAppendStream|toRetractStream)`
+have been deprecated. Use the `(from|to)(Data|Changelog)Stream` alternatives introduced in 1.13.
+
+#### Remove old connectors and formats stack around descriptors
+
+##### [FLINK-23513](https://issues.apache.org/jira/browse/FLINK-23513)
+
+The legacy versions of the SQL Kafka connector and SQL Elasticsearch connector have been removed
+together with their corresponding legacy formats. DDL or descriptors that still use `'connector.type='` or
+`'format.type='` options need to be updated to the new connector and formats available via the `'connector='` option.
+
+#### Drop BatchTableSource/Sink HBaseTableSource/Sink and related classes
+
+##### [FLINK-22623](https://issues.apache.org/jira/browse/FLINK-22623)
+
+The HBaseTableSource/Sink and related classes including various HBaseInputFormats and
+HBaseSinkFunction have been removed. It is possible to read via Table & SQL API and convert the
+Table to DataStream API (or vice versa) if necessary. DataSet API is not supported anymore.
+
+#### Drop BatchTableSource ParquetTableSource and related classes
+
+##### [FLINK-22622](https://issues.apache.org/jira/browse/FLINK-22622)
+
+The ParquetTableSource and related classes including various ParquetInputFormats have been removed.
+Use the filesystem connector with a Parquet format as a replacement. It is possible to read via
+Table & SQL API and convert the Table to DataStream API if necessary. DataSet API is not supported
+anymore.
+
+#### Drop BatchTableSource OrcTableSource and related classes
+
+##### [FLINK-22620](https://issues.apache.org/jira/browse/FLINK-22620)
+
+The OrcTableSource and related classes (including OrcInputFormat) have been removed. Use the
+filesystem connector with an ORC format as a replacement. It is possible to read via Table & SQL API
+and convert the Table to DataStream API if necessary. DataSet API is not supported anymore.
+
+#### Drop usages of BatchTableEnvironment and old planner in Python
+
+##### [FLINK-22619](https://issues.apache.org/jira/browse/FLINK-22619)
+
+The Python API does not offer a dedicated BatchTableEnvironment anymore. Instead, users can switch
+to the unified TableEnvironment for both batch and stream processing. Only the Blink planner (the
+only remaining planner in 1.14) is supported.
+
+#### Migrate ModuleFactory to the new factory stack
+
+##### [FLINK-23720](https://issues.apache.org/jira/browse/FLINK-23720)
+
+The `LOAD/UNLOAD MODULE` architecture for table modules has been updated to the new factory stack of
+FLIP-95. Users of this feature should update their `ModuleFactory` implementations.
+
+#### Migrate Table API to new KafkaSink
+
+##### [FLINK-23639](https://issues.apache.org/jira/browse/FLINK-23639)
+
+Table API/SQL write to Kafka with the new KafkaSink.
```

Comment:
@fapaul @AHeise May I ask you to help me with this one?

If I understand it correctly the state does not play any role here as it is for Table API? What are the user facing consequences? Has the properties changed? If not, do we even need to mention it?

Reply:
```suggestion
Table API/SQL write to Kafka with the new `KafkaSink`. When migrating from a query writing to Kafka in exactly-once mode from an earlier Flink version, make sure to terminate the old application with stop-with-savepoint to avoid lingering Kafka transactions. To run in exactly-once processing mode, the sink needs a user-configured and unique transaction prefix, such that transactions of different applications do not interfere with each other. 
```